public.tf

```diff
@@ -21,11 +23,15 @@ resource "aws_subnet" "public" {
   availability_zone = "${element(var.availability_zones, count.index)}"
   cidr_block        = "${cidrsubnet(signum(length(var.cidr_block)) == 1 ? var.cidr_block : data.aws_vpc.default.cidr_block, ceil(log(length(data.aws_availability_zones.available.names) * 2, 2)), length(data.aws_availability_zones.available.names) + count.index)}"
 
-  tags = {
-    "Name"      = "${module.public_subnet_label.id}${var.delimiter}${replace(element(var.availability_zones, count.index),"-",var.delimiter)}"
-    "Stage"     = "${module.public_subnet_label.stage}"
-    "Namespace" = "${module.public_subnet_label.namespace}"
-  }
+  tags = "${
+    merge(
+      map(
+        "Name", "${module.public_subnet_label.id}${var.delimiter}${replace(element(var.availability_zones, count.index),"-",var.delimiter)}",
```

Comment:
Is there some way we could modify `terraform-null-label` and add a variable called `ordinals`. This would then indicate that we want N labels. Each of those labels would get an attribute corresponding to the `count.index` position in the `ordinal` list. 

In this case we could pass `availability_zones` as an ordinal and then maybe we output an array of `id` fields.  I'm not sure how we would handle the `tags` map. I seem to recall we had trouble returning a list of maps.

Reply:
It is an attempt to make direct connection between `count`, `availability_zone` name and `Name` tag.

What would be if we define a module of `labels` with AZ names and `count` and use it with `aws_subnet` resource's `count`?
If array of `label.ids` was modified (e.g. automatically sorted), we will get improper names assigned to the `subnets`.

Reply:
> It is an attempt to make direct connection between count, availability_zone

This is what I'm calling an `ordinal`. I think that's a mapping between numbers (e.g. 0, 1, 2, 3) and words (`one`, `two, `three`... or in this case `us-east-1a`, `us-east-1b`, etc...). Is there a better word for that? Maybe `ordinal` is not the right word

Reply:
I think, we have corresponding task in the `Backlog` already.
https://app.asana.com/0/275899375677150/413199533841835
@osterman @aknysh What do you think of this strategy?